q2_diversity/_alpha/alpha_rarefaction_assets/src/state.js

```diff
@@ -0,0 +1,63 @@
+/* global d */
+import { setupData } from './data';
+import render from './render';
+
+function updateData(metric, category, svg, href, legend, legendTitle) {
+  href.attr('href', `${metric}.csv`);
+  let data = d[metric];
+  if (category) {
+    data = d[metric][category];
+  }
+  const preppedData = setupData(data, metric);
+  render(svg, preppedData, category, legend, legendTitle);
+}
+
+let curState = null;
+class State {
+  constructor() {
+    if (!curState) {
+      curState = this;
+    }
```

Comment:
It seems like this is trying to be a singleton, but I think it's backwards, you want to return curState if it was already defined. In either case, since `state` get's exported at the bottom, I don't think we need to worry about `curState` at all. So these parts can just be removed.